WordPress/gutenberg
# More intuitive Details block with summary and innerBlocks content

Simplifies the experimental Details block to have the `<summary>` as part of the block itself and Innerblocks for the content, suggested by @talldan in https://github.com/WordPress/gutenberg/pull/45055#issuecomment-1478983941. This is the same method used in the Quote block as well. I'd feel comfortable moving this out of the experiment following this PR. 

### This method makes the Details block simpler by: 
- **Reducing the complexities around style conflicts, as there is one block with design tooling—rather than three.** Noted in https://github.com/WordPress/gutenberg/pull/45055#issuecomment-1468863344). 
- **Allowing the Details block itself to be easily selectable.** In trunk, you have to first select the child "Details Summary" or "Details Content" blocks, then select the parent from the toolbar. 
- **Removing an extraneous tree level of blocks**. as the parent block _is_ the Innerblocks. 
- Negating the need for block locking (and therefore additional icons), as the summary will always be at the top of the Details block.

## What?
1. Removes the Details Summary and Details Content blocks.
2. Moves InnerBlocks capability to the Details block.
3. Tidies up the initial placeholder state and block example.
4. Improves the details icon to better support hi-res devices.
5. Simplifies the block description.

## Testing Instructions

1. Enable the Details block experiment within Gutenberg admin settings.
2. Open a Post or Page.
3. Insert a Details block. 
3. Add a summary.
4. Add blocks to the Details block content.
5. Leverage design tooling on the Details block.
6. See changes on the front-end. 

## Visuals

#### Example, icon, and description: 
<img width="686" alt="CleanShot 2023-04-13 at 13 43 59" src="https://user-images.githubusercontent.com/1813435/231841562-dda470e0-dd9d-4884-880e-82e3dc86d0e8.png">

#### Empty:
<img width="704" alt="CleanShot 2023-04-13 at 13 38 45" src="https://user-images.githubusercontent.com/1813435/231840358-d62767df-9eac-4c1e-b887-cab4fb8abe32.png">

#### Completed: 
<img width="717" alt="CleanShot 2023-04-13 at 13 40 52" src="https://user-images.githubusercontent.com/1813435/231840810-883d65d3-d1ee-484e-aad3-41100ddd02b0.png">

#### Completed with additional styling (to show how it can still be flexed using this method): 
<img width="1110" alt="CleanShot 2023-04-13 at 13 42 44" src="https://user-images.githubusercontent.com/1813435/231841165-a806d062-7d20-4a04-ac3c-74a0f31b4ff8.png">
<img width="703" alt="CleanShot 2023-04-13 at 13 49 12" src="https://user-images.githubusercontent.com/1813435/231842606-57362e25-a6a0-46e9-8363-a8bb4badbe33.png">
<img width="713" alt="CleanShot 2023-04-13 at 13 58 53" src="https://user-images.githubusercontent.com/1813435/231844807-6f370ded-48f5-49d1-b179-635096cec263.png">
<img width="690" alt="CleanShot 2023-04-13 at 14 00 00" src="https://user-images.githubusercontent.com/1813435/231844948-fc949f7d-b7e8-4b86-91f3-ecc7438fbf87.png">
